### PR TITLE
Update docs and version to 0.3

### PR DESCRIPTION
**Pull request by felipemorais**
Let's start discussing.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -47,98 +47,106 @@
...
-
-<p><em>uri:</em> A relative URI to a local script to load.</p>
+<p><strong>basket.require( objects )</strong></p>
+
+<p><em>objects:</em> One or more plain JavaScript objects to configure the require operation.</p>
+<dl>
+	<dt>url</dt>
+		<dd><strong>Required</strong> Relative URI to a local script.</dd>
+	<dt>key</dt>
+		<dd><strong>Optional</strong> Use this to define an alternative name, will default to <em>uri.</em></dd>
```
Comment (sindresorhus) [inline index.html:67]: internally

File: index.html [hunk 2]
```diff
@@ -47,98 +47,106 @@
...
+	<dt>expire</dt>
+		<dd><strong>Optional</strong> Set an expiration date, defined in hours before basket.js will re-fetch the script.</dd>
```
Comment (sindresorhus) [inline index.html]: lifespan of file cache in localStorage in hours

File: index.html [hunk 3]
```diff
@@ -47,98 +47,106 @@
...
+	<dt>execute</dt>
+		<dd><strong>Optional</strong> Whether to execute the script on load.</dd>
```
Comment (sindresorhus) [inline index.html]: Defaults to 'true'. Set to 'false' to only cache, but not execute the file.

File: index.html [hunk 4]
```diff
@@ -47,98 +47,106 @@
...
+	<dt>unique</dt>
+		<dd><strong>Optional</strong> used to set a version id to file.</dd>
```
Comment (sindresorhus) [inline index.html]: used to bust a cached file

File: index.html [hunk 5]
```diff
@@ -47,98 +47,106 @@
...
+<p>Similarly, if you wish to store <code>jquery-1.7.1.min.js</code> as <code>jquery</code>, this can be done as well:</p>
 
-<p><strong>basket.add( uri, [,options], [,callback] )</strong></p>
+<pre><code>basket.require({ url: 'jquery-1.7.1.min.js', key: 'jquery' });</code></pre>
 
-<p><em>uri:</em> A relative URI to a local script to be saved to localStorage (but not executed)</p>
+<p><strong> If <code>key</code> is unset, the url will be used as key.</strong></p>
 
-<p><em>options:</em> An object allowing you to configure the add operation. It supports two optional keys: overwrite - A boolean defining whether an entry being added can overwrite an existing entry with the same path/filename.</p>
+<p>You can Use <code>expire</code> to set a lifetime in hours for a cached file. The script will be fetched again if required after this time.</p>
```
Comment (sindresorhus) [inline index.html]: use

File: index.html [hunk 6]
```diff
@@ -47,98 +47,106 @@
...
-});
+<p><code>get()</code> will return an object if script is found or false if not</p>
+<p>The object retuned will contains all the properties set to require and some other properties such as.</p>
+<dl>
+	<dt>stamp</dt>
+		<dd>The timestamp for when the file was fetched.</dd>
+	<dt>expire</dt>
+		<dd>Updated to the expiration timestamp.</dd>
+	<dt>data</dt>
+		<dd>Contains the script text.</dd>
```
Comment (sindresorhus) [inline index.html]: File contents

**Comment by sindresorhus** [comment]
@addyosmani

**Comment by addyosmani** [comment]
I agree with the suggested changes so far but will take another look and see if we're missing anything. Thanks for the inline comments @peol and @sindresorhus and for kick-starting this off @felipemorais :)

File: index.html (hunk 2 above)
Comment (felipemorais) [inline index.html]: "Set a lifespan of file cache in localStorage in hours" ?

Comment (addyosmani) [inline index.html]: We should be providing an example of this after "hours" (e.g re-fetching after 7 days = 7 \* 24 = 168)

**Comment by felipemorais** [comment]
@sindresorhus @peol  @addyosmani #45 Thoughts?

**Comment by peol** [comment]
Looks good to me. I'll merge if no one has any objections later today :)

**Comment by addyosmani** [comment]
On 45, I think we should wait until 0.4 to address moving towards something more inline with Promise spec. Otherwise this looks good to me :) @sindresorhus anything more to add?

**Comment by sindresorhus** [comment]
nop

**Comment by felipemorais** [comment]
wait() can be assigned to an obj in require(). We really need this? it isn't documented.

```
basket.require( {
        url: 'fixtures/jquery.min.js' ,
        wait: function() {...}
} );
```

**Comment by addyosmani** [comment]
Merged. Lets create new issues to continue discussions :)

**Comment by sindresorhus** [comment]
@addyosmani Should we tag the 0.3 release and announce it, and put it in Bower?

**Comment by addyosmani** [comment]
I think we should tag the release, come up with an updated changelog, put it on Bower and announce on Monday. How does that sound?

**Comment by peol** [comment]
Sounds good to me.

**Comment by sindresorhus** [comment]
Sgtm2

**Pull request merged**